dln/operator.py

```diff
@@ -361,21 +378,66 @@ def instantiate_tokenizer(model_name: str):
         encoder = AutoTokenizer.from_pretrained(pretrained_path)
     return encoder
 
-class LLMRegistry:
-    def __init__(self):
-        self.models = []
 
-    def register(self, model_name: str, **generation_options) -> LLM:
-        if model_name in GPT.AVAILABLE_MODELS:
-            llm = GPT(model_name, **generation_options)
-        else:
-            llm = VLLM(model_name, **generation_options)
+class LLMRegistry:
 
-        self.models.append(llm)
+    def __init__(self, config=None):
+        self.models : Dict[str, LLM] = {}
+        if config is not None:
+            self._load_from_configs(config)
+
+    def register(self, model_name: str, model_type: str = None, **generation_options) -> LLM:
+        """ Register a single model to the registry.
+        Args:
+            model_name: how you refer to the model, for example: gpt-3.
+            model_type: the api model name, for example: text-davinci-003. If not provided, use model_name as default.
+            **generation_options: generation options, for example: api_key, api_base, api_type, api_version, max_tokens, temperature, etc.
+        Returns:
+            the instantiated model
+        """
+        if model_type is None:
+            model_type = model_name
+        llm = self.instantiate_llm(model_type, **generation_options)
+        self.models[model_name] = llm
         return llm
 
     @property
     def total_cost(self):
-        return sum([llm.total_cost for llm in self.models])
+        return sum([llm.total_cost for llm in self.models.values()])
+
+    @classmethod
+    def instantiate_llm(cls, model_type: str, **generation_options) -> LLM:
```

Comment:
What is the purpose of having both register and instantiate_llm?

Reply:
I can remove `instantiate_llm` if it is confusing. This function is just a syntactic sugar, so you don't need to instantiate the registry to get an LLM.

Reply:
We could make one singleton instance of the Registry.

Reply:
I would be fine if we didn't implement the singleton for now. What do you think?